acceptance/tests/puppet_device_test.rb

```diff
@@ -1,74 +1,73 @@
+  # configure device credentials
+  on(master, puppet("apply -e 'device_manager { [\"spinny1.example.com\", \"spinny2.example.com\", \"spinny3.example.com\"]: type => 'spinner', credentials => { facts_cpu_time => 1 }, include_module => false }'"))
 
-# run operation without certificate
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com --facts")
+  # run operation without certificate
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com --facts")
 
-# request cert, or fall through
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com", acceptable_exit_codes: [0, 1]) do |result|
-  assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed')
-end
+  # request cert, or fall through
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com", acceptable_exit_codes: [0, 1]) do |result|
+    assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed')
+  end
 
-on(master, "umask 077; puppetserver ca sign --certname spinny1.example.com")
+  on(master, "umask 077; puppetserver ca sign --certname spinny1.example.com")
 
-# test catalog application
-on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com")
+  # test catalog application
+  on(master, "umask 077; puppet device #{common_options} --target spinny1.example.com")
 
-# test --resource
-on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --resource spinner")
+  # test --resource
+  on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --resource spinner")
 
-# test --apply
-on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --apply /tmp/spinner.pp")
+  # test --apply
+  on(master, "umask 077; puppet device  #{common_options} --target spinny1.example.com --apply /tmp/spinner.pp")
 
-# test development runmode
-common_options += " --libdir #{codedir}/test_device/lib"
-on(master, "umask 077; puppet device #{common_options} --target spinny2.example.com --facts")
-on(master, "umask 077; puppet device #{common_options} --target spinny2.example.com", acceptable_exit_codes: [0, 1]) do |result|
-  assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed with --libdir')
+  # test development runmode
+  common_options += " --libdir #{codedir}/test_device/lib"
+  on(master, "umask 077; puppet device #{common_options} --target spinny2.example.com --facts")
+  on(master, "umask 077; puppet device #{common_options} --target spinny2.example.com", acceptable_exit_codes: [0, 1]) do |result|
+    assert_no_match(/Permission denied/, result.stderr, 'cert requesting failed with --libdir')
+  end
+  on(master, "umask 077; puppetserver ca sign --certname spinny2.example.com")
```

Comment:
I read the ticket, but I still don't get why we need the umask. Though I don't think we'd ever need it with this command???

Reply:
@tphoney ?

Reply:
I put it there, because "professional paranoia". When the user is running all commands under `umask`, then all commands in the test should run under the same umask. Who knows what'll shake out.